### PR TITLE
use unprivileged nginx as base image / dockerfile

### PR DESCRIPTION
**Pull request by PhilippHeuer**
### Description

This changes the base image for the Dockerfile from `nginx:1.19-alpine` to `nginxinc/nginx-unprivileged:1.18-alpine` and removes the modified nginx.conf as it's content is equal to that of the unprivileged nginx default config. (including that the nginx version will not be exposed)

I also replaced the MAINTAINER instruction with the author label as MAINTAINER is deprecated since docker `1.13`.

### Motivation and Context

There are a lot of resources available on why running containers as root is not recommended -> https://amazicworld.com/get-the-evil-out-dont-run-containers-as-root/#:~:text=One%20of%20the%20key%20arguments,services%2C%20create%20users%2C%20etc.

Some container orchestrators (ie. openshift) will deny running containers as root and instead assign random user ids to execute workloads (which will not work with the current image). So this change makes the official swagger-editor image usable for more users, as it will work regardless of which user id is used to start the container.

### How Has This Been Tested?

I build the image with v3.14.7 and verified that everything works by using the website.

### My PR contains... 

- [x] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [x] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,6 +12,12 @@ ADD ./dist/*.css /usr/share/nginx/html/dist/
 ADD ./dist/*.png /usr/share/nginx/html/dist/
 ADD ./docker-run.sh /usr/share/nginx/
 
+# needed for gzip after replacements / SWAGGER_FILE
+RUN chgrp -R 0 /usr/share/nginx/html && \
+    chmod -R g=u /usr/share/nginx/html
```
Comment (mrsegen) [inline Dockerfile:17]: I don't assert that there is anything wrong with this approach. However, I note that it is a different pattern than what is used in the [Dockerfile of the swagger-ui repo](https://github.com/swagger-api/swagger-ui/blob/887f45ff1a3da350e7dfda7420926a145e341b72/Dockerfile#L24). 

It's not my call, but I _imagine_ Swagger would prefer to keep these consistent.

Full disclosure: I'm in the biased position of having just put up a [PR with a conflicting change](https://github.com/swagger-api/swagger-editor/pull/2281). It depends on the nginx.conf ([spot 1](https://github.com/swagger-api/swagger-editor/pull/2281/files#diff-360c358e5fc4de52699ff11376043b7c67ebccc164c8dd9bc2b025782d681279R10-R12), [spot 2](https://github.com/swagger-api/swagger-editor/pull/2281/files#diff-360c358e5fc4de52699ff11376043b7c67ebccc164c8dd9bc2b025782d681279R27-R29)) to enable altering the BASE_URL via environment variable.

**Review by mrsegen — COMMENTED** [review]
I appreciate the movement towards the unprivileged build.

My comment on the Dockerfile speaks to my concern regarding https://github.com/swagger-api/swagger-editor/pull/2281 depending on the nginx.conf removed by this PR.

**Comment by mrsegen** [comment]
@tim-lai @PhilippHeuer responding to the comment on #2281 (but pertaining more to this this PR):

Indeed `1.18` is currently tagged as the `stable` image for nginx: https://hub.docker.com/_/nginx.

This may be a naive question, but: How much would _just_ reverting to 1.18 help?

**Comment by tim-lai** [comment]
@PhilippHeuer Closing in favor of #2281. Thanks for the contribution!

**Pull request closed without merging**